resolwe_bio/processes/reads_processing/bbduk.yml

```diff
@@ -0,0 +1,655 @@
+# =============================================
+# Processors for reads procesing - BBduk
+# =============================================
+#
+# Processors for NGS reads filtering/trimming
+---
+
+- slug: bbduk-single
+  name: BBDuk from BBMap 36.20 (single-end)
+  requirements:
+    expression-engine: jinja
+  data_name: BBduk'{{ reads.fastq.0.file|default("?") }}'
+  version: 0.0.1
+  type: data:reads:fastq:single:bbduk
+  category: analyses
+  flow_collection: sample
+  persistence: CACHED
+  description: >
+    BBduk is capable of quality-trimming and filtering, adapter-trimming, contaminant-filtering via
+    kmer matching, sequence masking, GC-filtering, length filtering, entropy-filtering, format
+    conversion, histogram generation, subsampling, quality-score recalibration, kmer cardinality
+    estimation, and various other operations in a single pass.
+  input:
+    - name: reads
+      label: Reads
+      type: data:reads:fastq:single
+    - name: adapters
+      label: Adapters
+      group:
+        - name: bbduk_adapters
+          label: bbduk files with adapters
+          type: basic:string
+          required: false
+          choices:
+            - label: Illumina Truseq DNA adapters.
+              value: '~/bbmap/resources/truseq.fa.gz'
```

Comment:
I am testing if this adapters are used in reads filtering and it is working correctly.